addons/wiki/migrations/0009_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,195 @@
+    return
+
+def update_comments_viewed_timestamp(node, current_wiki_object, desired_wiki_object):
+    """Replace the current_wiki_object keys in the comments_viewed_timestamp dict with the desired wiki_object_id """
+    contributors_pending_save = []
+    for contrib in node.contributors.exclude(comments_viewed_timestamp={}):
+        if contrib.comments_viewed_timestamp.get(current_wiki_object._id, None):
+            timestamp = contrib.comments_viewed_timestamp[current_wiki_object._id]
+            contrib.comments_viewed_timestamp[desired_wiki_object._id] = timestamp
+            del contrib.comments_viewed_timestamp[current_wiki_object._id]
+            contributors_pending_save.append(contrib)
+    bulk_update(contributors_pending_save, batch_size=1000)
+    return
+
+def migrate_guid_referent(guid, desired_referent, content_type_id):
+    """
+    Point the guid towards the desired_referent.
+    Pointing the NodeWikiPage guid towards the WikiPage will still allow links to work.
+    """
+    guid.content_type_id = content_type_id
+    guid.object_id = desired_referent.id
+    return guid
+
+def create_wiki_page(node, node_wiki, page_name):
+    wp = WikiPage(
+        page_name=page_name,
+        user_id=node_wiki.user_id,
+        node=node,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+    )
+    wp.update_modified = False
+    return wp
+
+def create_wiki_version(node_wiki, wiki_page):
+    wv = WikiVersion(
+        wiki_page=wiki_page,
+        user_id=node_wiki.user_id,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+        content=node_wiki.content,
+        identifier=node_wiki.version,
+    )
+    wv.update_modified = False
+    return wv
+
+def create_guids():
+    content_type = ContentType.objects.get_for_model(WikiPage)
+    progress_bar = progressbar.ProgressBar(maxval=WikiPage.objects.count()).start()
+    logger.info('Creating new guids for all WikiPages:')
+    for i, wp in enumerate(WikiPage.objects.all(), 1):
+        # looping instead of bulk_create, so _id's are not the same
+        progress_bar.update(i)
+        Guid.objects.create(object_id=wp.id, content_type_id=content_type.id)
+    progress_bar.finish()
+    logger.info('WikiPage guids created.')
+    return
+
+def create_wiki_pages(nodes):
+    wiki_pages = []
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    logger.info('Starting migration of WikiPages:')
+    for i, node in enumerate(nodes, 1):
+        progress_bar.update(i)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                node_wiki = NodeWikiPage.objects.filter(former_guid=version_list[0]).only('user_id', 'date', 'modified').get()
+                latest_page_name = NodeWikiPage.objects.filter(former_guid=version_list[-1]).values_list('page_name', flat=True).get()
```

Comment:
Even though you're using `only`, we're still auto-including the `guid`. 

```
LOG:  duration: 145.897 ms  statement: SELECT "addons_wiki_nodewikipage"."page_name", (SELECT JSON_AGG(JSON_BUILD_ARRAY(U0."modified", U0."id", U0."_id", U0."content_type_id", U0."object_id", U0."created") ORDER BY U0."created" DESC) AS "__fields" FROM "osf_guid" U0 WHERE (U0."object_id" = addons_wiki_nodewikipage."id") AND (U0."content_type_id" = 108)) AS "__guids" FROM "addons_wiki_nodewikipage" WHERE "addons_wiki_nodewikipage"."former_guid" = '<snip>'
```

You should be able toavoid the GUID queries by doing `.include(None)`. Same goes with the line before this.

Reply:
ahhh, nice @sloria, with this change I had my fastest run yet, down to 3 min 50 seconds on staging data.